rune/libenclave/internal/runtime/libenclave_v1.proto

```diff
@@ -0,0 +1,96 @@
+syntax = "proto3";
+
+package runtime_proto;
+
+service Libenclave {
+	rpc GetVersion(google.protobuf.Empty) returns (GetVersionResponse);
+	rpc GetCapability(google.protobuf.Empty) returns (GetCapabilityResponse);
+	rpc Create(CreateRequest) returns (CreateResponse);
+	rpc Delete(DeleteRequest) returns (google.protobuf.Empty);
+	rpc Init(InitRequest) returns (InitResponse);
+	rpc Spawn(SpawnRequest) returns (SpawnResponse);
+	rpc Exec(ExecRequest) returns (ExecResponse);
+	rpc Kill(KillRequest) returns (KillResponse);
+	rpc Attest(AttestRequest) returns (AttestResponse);
+}
+
+message GetVersionResponse {
+	uint32 version = 1;
+}
+
+message GetCapabilityResponse {
+	string capability = 1;
+}
+
+message CreateRequest {
+	string loglevel = 1;
+	string bundle = 2;
+	string stdin = 3;
+	string stdout = 4;
+	string stderr = 5;
+	string cpus = 6;
+	string memory = 7;
+	string kernel = 8;
+	string vsock = 9;
+}
+
+message CreateResponse {
+	string id = 1;
+	string status = 2;
+}
+
+message DeleteRequest {
+	string id = 1;
+}
+
+message InitRequest {
+	string id = 1;
+}
+
+message InitResponse {
+	string status = 1;
+}
+
+message SpawnRequest {
+	string id = 1;
+	string args = 2;
+}
+
+message SpawnResponse {
+	uint32 pid = 1;
+}
+
+message ExecRequest {
+	string id = 1;
+	uint32 pid = 2;
+	string filename = 3;
+	string argv = 4;
+	string envp = 5;
+}
+
+message ExecResponse {
+	string status = 1;
+}
+
+message KillRequest {
+	string id = 1;
+	uint32 pid = 2;
+	uint32 sig = 3;
+}
+
+message KillResponse {
+	string status = 1;
+}
+
+message AttestRequest {
+	string id = 1;
+	bool isRA = 2;
+	string spid = 3;
+	string subscriptionKey = 4;
+	string quoteType = 5;
```

Comment:
I think we need a ra type parameter instead of the details. Spid, subkey and quoteType all are specific to EPID remote attestation for SGX.